src/RequestBuilder.php

```diff
@@ -201,7 +201,19 @@ private function constructBodyAndQueryParameters(Message $message, array $config
                 $requiredQueryParam = Serializer::toCamelCase($requiredQueryParam);
                 if (!array_key_exists($requiredQueryParam, $queryParams)) {
                     $getter = Serializer::getGetter($requiredQueryParam);
-                    $queryParams[$requiredQueryParam] = $message->$getter();
+                    $queryParamValue = $message->$getter();
+                    if ($queryParamValue instanceof Message) {
+                        $param = json_decode($queryParamValue->serializeToJsonString(), true);
+                        if (is_array($param)) {
+                            foreach ($param as $key => $value) {
+                                $queryParams[$requiredQueryParam . '.' . $key] = $value;
```

Comment:
Can you add a comment that indicates this only supports one-level of nesting? i.e. it will not support enumerating each field of a message type field within this query param value (that is an instance of a Message).

Reply:
Added!